flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/catalog/DatabaseCalciteSchema.java

```diff
@@ -136,7 +136,7 @@ private Table convertTemporaryTable(
 				return convertSinkTable(connectorTable);
 			}
 		} else if (table instanceof CatalogTable) {
-			return convertCatalogTable(tablePath, (CatalogTable) table, null);
+			return convertCatalogTable(tablePath, (CatalogTable) table, null, TableStats.UNKNOWN);
```

Comment:
Is is possible for temporary tables also have table stats? 
@dawidwys What do you think about this?

Reply:
Here if we take a look to `extractTableStats`. It gets stats from catalog. I think `ConnectorCatalogTable` should never be put into catalog.
So I think we need remove `extractTableStats` of `ConnectorCatalogTable`.

Reply:
Yes, this is current implementation. But from user's perspective, when they registered a temporary table, it should also have an opportunity to have table stats.

Reply:
But the stats of temporary table should not come from catalog, right?
Another thing is that now temporary table is `CatalogTable` too.
`ConnectorCatalogTable` only come from deprecated `registerTableSource`.

Reply:
That is also my concern: users could also store  a `ConnectorCatalogTable` instance into a `Catalog` by `Catalog#createTable` (not by `TableEnvironment#fromTableSource` or `TableEnvironment#registerTableSource`) if the `ConnectorCatalogTable` instance is serializable (`GenericInMemoryCatalog` even does not need serializable object). should we prevent users from storing `ConnectorCatalogTable` instance into a `Catalog` ? or should we clear that `ConnectorCatalogTable` could only be used for temporary table ?

Reply:
In old planner, `ConnectorCatalogTable` may be a permanent table(from catalog) too in `DatabaseCalciteSchema`.